test/FixedStakingRewards.t.sol

```diff
@@ -998,4 +1074,420 @@ contract FixedStakingRewardsTest is Test {
+    }
+
+    function test_Stake_RevertWhen_Paused() public {
+        // Set up rewards and whitelist
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(1000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        // Pause the contract
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+    }
+
+    function test_Withdraw_RevertWhen_Paused() public {
+        // Set up staking first
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(2000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+
+        // Release rewards and pause
+        stakingRewards.releaseRewards();
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.withdraw(50e18);
+        vm.stopPrank();
+    }
+
+    function test_GetReward_RevertWhen_Paused() public {
+        // Set up staking and rewards
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(1000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+
+        // Move time forward and release rewards
+        skip(3600);
+        stakingRewards.releaseRewards();
+
+        // Pause the contract
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.getReward();
+        vm.stopPrank();
+    }
+
+    function test_Exit_RevertWhen_Paused() public {
+        // Set up staking first
+        stakingRewards.setRewardYieldForYear(1e18);
+        stakingRewards.supplyRewards(1000e18);
+        stakingRewards.addToWhitelist(user1);
+
+        vm.startPrank(user1);
+        stakingToken.approve(address(stakingRewards), 100e18);
+        stakingRewards.stake(100e18);
+        vm.stopPrank();
+
+        // Release rewards and pause
+        stakingRewards.releaseRewards();
+        stakingRewards.pause();
+
+        vm.startPrank(user1);
+        vm.expectRevert(abi.encodeWithSelector(Pausable.EnforcedPause.selector));
+        stakingRewards.exit();
+        vm.stopPrank();
+    }
+
+    function test_OwnerFunctions_WorkWhen_Paused() public {
+        // Owner functions should still work when paused
+        stakingRewards.pause();
+
+        // These should all work
+        stakingRewards.addToWhitelist(user1);
```

Comment:
do we actually need to go so far as to have this kind of testing? (effectively verifying that undesired functionality is *not* in the contract)